File: app/src/main/java/com/vtrack/feature/maintenance/history/MaintenanceHistoryScreen.kt
package com.vtrack.feature.maintenance.history

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.List
import androidx.compose.material3.Badge
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FloatingActionButton
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.navigation.NavController
import com.vtrack.data.model.MaintenanceRecord
import com.vtrack.util.FormatUtil
import com.vtrack.util.MaintenanceStatus
import com.vtrack.util.MaintenanceUrgency
import kotlin.math.abs

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun MaintenanceHistoryScreen(
    navController: NavController,
    onLogMaintenance: (Long, Long) -> Unit,
    viewModel: MaintenanceHistoryViewModel = hiltViewModel()
) {
    val uiState by viewModel.uiState.collectAsState()

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text(text = uiState.typeName.ifEmpty { "History" }) },
                navigationIcon = {
                    IconButton(onClick = { navController.popBackStack() }) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                }
            )
        },
        floatingActionButton = {
            if (uiState.vehicleId > 0) {
                FloatingActionButton(
                    onClick = { onLogMaintenance(uiState.vehicleId, uiState.typeId) }
                ) {
                    Icon(
                        imageVector = Icons.Default.Add,
                        contentDescription = "Log service"
                    )
                }
            }
        }
    ) { innerPadding ->
        when {
            uiState.isLoading -> {
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(innerPadding),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator()
                }
            }

            else -> {
                LazyColumn(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(innerPadding),
                    contentPadding = PaddingValues(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    // Status card at top
                    uiState.status?.let { status ->
                        item(key = "status_card") {
                            StatusCard(status = status)
                        }
                    }

                    if (uiState.records.isEmpty()) {
                        item(key = "empty") {
                            EmptyHistoryState()
                        }
                    } else {
                        item(key = "records_header") {
                            Text(
                                text = "Service History",
                                style = MaterialTheme.typography.titleSmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant,
                                modifier = Modifier.padding(top = 4.dp)
                            )
                        }

                        items(
                            items = uiState.records,
                            key = { it.id }
                        ) { record ->
                            ServiceRecordCard(record = record)
                        }
                    }
                }
            }
        }
    }
}

@Composable
private fun StatusCard(status: MaintenanceStatus) {
    val urgencyColor = when (status.urgency) {
        MaintenanceUrgency.OK -> Color(0xFF4CAF50)
        MaintenanceUrgency.DUE_SOON -> Color(0xFFFFC107)
        MaintenanceUrgency.OVERDUE -> Color(0xFFF44336)
    }

    val urgencyLabel = when (status.urgency) {
        MaintenanceUrgency.OK -> "OK"
        MaintenanceUrgency.DUE_SOON -> "Due Soon"
        MaintenanceUrgency.OVERDUE -> "Overdue"
    }

    val urgencyTextColor = when (status.urgency) {
        MaintenanceUrgency.OK -> Color.White
        MaintenanceUrgency.DUE_SOON -> Color.Black
        MaintenanceUrgency.OVERDUE -> Color.White
    }

    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceVariant
        )
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Current Status",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.SemiBold
                )
                Badge(
                    containerColor = urgencyColor,
                    contentColor = urgencyTextColor
                ) {
                    Text(
                        text = urgencyLabel,
                        modifier = Modifier.padding(horizontal = 4.dp)
                    )
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            Text(
                text = if (status.milesUntilDue >= 0) {
                    "${FormatUtil.formatMiles(status.milesUntilDue)} until due"
                } else {
                    "Overdue by ${FormatUtil.formatMiles(abs(status.milesUntilDue))}"
                },
                style = MaterialTheme.typography.bodyLarge,
                color = if (status.urgency == MaintenanceUrgency.OVERDUE) {
                    MaterialTheme.colorScheme.error
                } else {
                    MaterialTheme.colorScheme.onSurfaceVariant
                }
            )

            Spacer(modifier = Modifier.height(8.dp))

            LinearProgressIndicator(
                progress = { status.percentUsed.coerceIn(0.0, 1.0).toFloat() },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(8.dp),
                color = urgencyColor,
                trackColor = urgencyColor.copy(alpha = 0.2f)
            )

            Spacer(modifier = Modifier.height(8.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Text(
                    text = "Every ${FormatUtil.formatMiles(status.type.intervalMiles)}",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Text(
                    text = "Current: ${FormatUtil.formatMiles(status.currentOdometer)}",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
private fun ServiceRecordCard(record: MaintenanceRecord) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = FormatUtil.formatDate(record.date),
                    style = MaterialTheme.typography.labelLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Text(
                    text = FormatUtil.formatMiles(record.odometer),
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.SemiBold
                )
            }

            if (record.cost != null || record.notes != null) {
                Spacer(modifier = Modifier.height(8.dp))
            }

            if (record.cost != null) {
                Text(
                    text = FormatUtil.formatCurrency(record.cost),
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )
            }

            if (record.notes != null) {
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = record.notes,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
private fun EmptyHistoryState() {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 48.dp),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center
        ) {
            Icon(
                imageVector = Icons.Default.List,
                contentDescription = null,
                modifier = Modifier.size(48.dp),
                tint = MaterialTheme.colorScheme.onSurfaceVariant
            )
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = "No service records yet",
                style = MaterialTheme.typography.bodyLarge,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center
            )
            Spacer(modifier = Modifier.height(4.dp))
            Text(
                text = "Tap + to log your first service",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center
            )
        }
    }
}

File: vtrack-ios/VTrack/Views/Maintenance/MaintenanceHistoryScreen.swift
import SwiftUI
import SwiftData

struct MaintenanceHistoryScreen: View {
    @Environment(\.modelContext) private var modelContext
    @State private var viewModel: MaintenanceHistoryViewModel?

    let typeId: String
    var onLogMaintenance: (String, String) -> Void

    var body: some View {
        Group {
            if let vm = viewModel {
                if vm.records.isEmpty && vm.status == nil {
                    EmptyStateView(
                        icon: "wrench.fill",
                        title: "No service records yet",
                        message: "Tap + to log your first service"
                    )
                } else {
                    List {
                        if let status = vm.status {
                            Section("Current Status") {
                                VStack(alignment: .leading, spacing: 8) {
                                    HStack {
                                        Text("Status")
                                            .font(.subheadline)
                                        Spacer()
                                        urgencyBadge(status.urgency)
                                    }
                                    Text(status.urgency == .overdue
                                         ? "Overdue by \(FormatUtil.formatMiles(abs(status.milesUntilDue)))"
                                         : "\(FormatUtil.formatMiles(status.milesUntilDue)) until due")
                                    .font(.subheadline)
                                    ProgressView(value: min(max(status.percentUsed, 0), 1))
                                        .tint(colorForUrgency(status.urgency))
                                    HStack {
                                        Text("Every \(FormatUtil.formatMiles(status.type.intervalMiles))")
                                        Spacer()
                                        Text("Current: \(FormatUtil.formatMiles(status.currentOdometer))")
                                    }
                                    .font(.caption)
                                    .foregroundStyle(.secondary)
                                }
                            }
                        }
                        if !vm.records.isEmpty {
                            Section("Service History") {
                                ForEach(vm.records, id: \.persistentModelID) { record in
                                    VStack(alignment: .leading, spacing: 4) {
                                        HStack {
                                            Text(FormatUtil.formatDate(record.date))
                                                .font(.subheadline)
                                                .fontWeight(.medium)
                                            Spacer()
                                            Text(FormatUtil.formatMiles(record.odometer))
                                                .font(.subheadline)
                                                .foregroundStyle(.secondary)
                                        }
                                        if let cost = record.cost {
                                            Text(FormatUtil.formatCurrency(cost))
                                                .font(.caption)
                                                .foregroundStyle(.secondary)
                                        }
                                        if let notes = record.notes, !notes.isEmpty {
                                            Text(notes)
                                                .font(.caption)
                                                .foregroundStyle(.secondary)
                                        }
                                    }
                                    .padding(.vertical, 2)
                                }
                            }
                        }
                    }
                }
            } else {
                ProgressView()
            }
        }
        .navigationTitle(viewModel?.typeName ?? "History")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            ToolbarItem(placement: .primaryAction) {
                Button {
                    if let vm = viewModel, let vehicleId = vm.vehicleId, let typeId = vm.typeId {
                        onLogMaintenance(vehicleId, typeId)
                    }
                } label: {
                    Image(systemName: "plus")
                }
            }
        }
        .onAppear {
            if viewModel == nil {
                viewModel = MaintenanceHistoryViewModel(modelContext: modelContext, typeId: typeId)
            }
            viewModel?.load()
        }
    }

    @ViewBuilder
    private func urgencyBadge(_ urgency: MaintenanceUrgency) -> some View {
        let (text, bg, fg): (String, Color, Color) = {
            switch urgency {
            case .overdue: return ("Overdue", .red, .white)
            case .dueSoon: return ("Due Soon", .orange, .black)
            case .ok: return ("OK", .green, .white)
            }
        }()
        Text(text)
            .font(.caption2)
            .fontWeight(.semibold)
            .padding(.horizontal, 6)
            .padding(.vertical, 2)
            .background(bg)
            .foregroundStyle(fg)
            .clipShape(Capsule())
    }

    private func colorForUrgency(_ urgency: MaintenanceUrgency) -> Color {
        switch urgency {
        case .overdue: return .red
        case .dueSoon: return .orange
        case .ok: return .green
        }
    }
}
